### PR TITLE
feat: update create command to support Intel GPU notebooks

### LINKED ISSUE #147: Update `dss create` command to support Intel GPU notebooks.
### Why it needs to get done

In order to create a kubernetes pod with the intel gpu acceleration enabled we must:
- Have the intel GPU operator enabled in Kubernetes cluster. 
- Have the Kubernetes resources section filled with the `gpu.intel.com/i915` section. Example [here](https://github.com/canonical/data-science-stack/blob/frenchwr/intel-gpu-integration/src/dss/manifest_templates/notebook_deployment.yaml.j2#L46). 

After discussing with the team we decided to drop the `--gpu intel` argument from the `dss create` command. If the Intel acceleration is enabled (by user manually deploying the intel gpu operator) , all the notebooks will have the intel resources section filled automatically. Meaning that having correct image notebooks can use Intel hardware. This is not the problem for images without intel librarries as they will not use the resource anyways. 

Because of this the `dss create` should check for the presence of intel gpu plugin. I f the plugin is there it will automatically populate the resources section.

Because we are using `intel/intel-extension-for-tensorflow:2.15.0-xpu-idp-jupyter` and `intel/intel-extension-for-pytorch:2.1.20-xpu-idp-jupyter` images for intel ML notebooks we also need to adjust the `command` and `args` section (check the [example](https://github.com/canonical/data-science-stack/blob/frenchwr/intel-gpu-integration/src/dss/manifest_templates/notebook_deployment.yaml.j2#L29)). We can add these settings blobally to all dss notebook deployments as non intel ones are setting these in their Dockerfiles anyways (This I need to test). 

We also need to add `intel/intel-extension-for-tensorflow:2.15.0-xpu-idp-jupyter` and `intel/intel-extension-for-pytorch:2.1.20-xpu-idp-jupyter` images are recommendations to `dss create --help`

You can find more information abou intel plugin here in this [spec](https://docs.google.com/document/d/1E6pZEJLEvlb63ZyAVOkwKPFdUgMJW2HZ1B4ypvECBoI/edit#heading=h.cqrgvanzectb).

### What needs to get done

- Add command and args section to every DSS notebook deployment to support intel nobooks without any changes in dss create. 
- Add resources section for intel devices if intel plugin is enabled 
- Align the design decisions of this change with the spec for Intel DSS
- Align the design decisions of this change with discussion with the UX team

### When is the task considered done

- `dss create` can schedule intel gpu images as notebooks namely `intel/intel-extension-for-tensorflow:2.15.0-xpu-idp-jupyter` and `intel/intel-extension-for-pytorch:2.1.20-xpu-idp-jupyter`.
- Aforementioned images are part of `dss create --help` list of recommended notebooks.
- Proper unit tests are written.

**Note to integration tests:** at the time of writing the Intel team should be responsible for handling the integration tests as we cant access the machines with intel hardware in CI. Please refer to the outcome of this [task](https://docs.google.com/document/d/1_EeQgdzj6NZ4eKi_Q738s-8mGdQlUjT3zMWWJQuY8fk/edit#heading=h.idu24lp0fpjw) for next steps.

### PR DESCRIPTION
**Pull request by DnPlas**
This commit introduces an automatic way of scheduling Notebook Servers on Nodes labeled as intel.feature.node.kubernetes.io/gpu.
This commit also affects the command and args that run for the Notebook Servers containers, as these are now part of the requirements for making Intel GPUs work. They should be exactly as [this](https://github.com/canonical/data-science-stack/blob/frenchwr/intel-gpu-integration/src/dss/manifest_templates/notebook_deployment.yaml.j2#L29C1-L43C39).

Fixes #147

### Manual testing

Assuming you have a microk8s cluster with hostpath-storage enabled

1. Clone this repository, checkout to the branch of this PR
2. Build and install from source `pip install .`
3. Initialise `dss initialize --kubeconfig="$(sudo microk8s config)"`
4. Label the node to simulate the Intel plugin has done its job `kubectl label node <name of your node> intel.feature.node.kubernetes.io/gpu=true`
5. Create a notebook `dss create my-notebook --image=ubuntu`
6. Verify the `Deployment` of the server has the resource limits we are interested in:

```
kubectl get deployment -ndss my-notebook -oyaml
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    ...
    spec:
      containers:
      - env:
        - name: MLFLOW_TRACKING_URI
          value: http://mlflow.dss.svc.cluster.local:5000
        image: kubeflownotebookswg/jupyter-pytorch-full:v1.8.0
        imagePullPolicy: IfNotPresent
        name: my-notebook
        ports:
        - containerPort: 8888
          name: notebook-port
          protocol: TCP
        resources:
          limits:
            gpu.intel.com/i915: "1" # <--- we are interested in this
```

7. Also verify that the `command` and `args` are always set:

```
          command:
            - jupyter
          args:
            - lab
            - --notebook-dir=/home/jovyan
            - --ip=0.0.0.0
            - --no-browser
            - --allow-root
            - --port=8888
            - --ServerApp.token=''
            - --ServerApp.password=''
            - --ServerApp.allow_origin='*'
            - --ServerApp.allow_remote_access=True
            - --ServerApp.authenticate_prometheus=False
            - --ServerApp.base_url='/'
```

Alternatively, you can try the create command w/o labelling the Node. In that case, the `Deployment` should not have any of the resource limits.

### CODE REVIEW

**Comment by misohu** [comment]
We need to add `command` and `args` section to all deployments according to the spec https://github.com/canonical/data-science-stack/compare/main...frenchwr/intel-gpu-integration#diff-e5f395a6247e35966f0a29978433e744bacd4913c9101d1ca46e364bdc249293

**Review by misohu — APPROVED** [review]
Tested on testflinger lab .... I was able to create intel ipex itex notebooks without problems ... I also tried pytorch without intel ... I tested case with and without intel plugin and looks like everything is ok
